integrations/symfony-bundle.rst

```diff
@@ -0,0 +1,176 @@
+| | httplug.plugin.decoder         | | These services are not public and may only be used when configure |
+| | httplug.plugin.error           | | HttpClients or other services.                                    |
+| | httplug.plugin.logger          |                                                                     |
+| | httplug.plugin.redirect        |                                                                     |
+| | httplug.plugin.retry           |                                                                     |
+| | httplug.plugin.stopwatch       |                                                                     |
++----------------------------------+---------------------------------------------------------------------+
+| | httplug.plugin.cache           | | These are plugins that are disabled by default.                   |
+| | httplug.plugin.cookie          | | They need to be configured before they can be used.               |
+| | httplug.plugin.history         | | These services are not public and may only be used when configure |
+|                                  | | HttpClients or other services.                                    |
++----------------------------------+---------------------------------------------------------------------+
+
+\* *These services are always an alias to another service. You can specify your own service or leave the default, which is the same name with `.default` appended. The default services in turn use the service discovery mechanism to provide the best available implementation. You can specify a class for each of the default services to use instead of discovery, as long as those classes can be instantiated without arguments.*
+
+
+If you need a more custom setup, define the services in your application configuration and specify your service in the `main_alias` section. For example, to add authentication headers, you could define a service that decorates the service `httplug.client.default` with a plugin that injects the authentication headers into the request and configure `httplug.main_alias.client` to the name of your service.
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            acme_client: # This is the name of the client
+            factory: 'httplug.factory.guzzle6'
+
+        main_alias:
+            client: httplug.client.default
+            message_factory: httplug.message_factory.default
+            uri_factory: httplug.uri_factory.default
+            stream_factory: httplug.stream_factory.default
+        classes:
+            # uses discovery if not specified
+            client: ~
+            message_factory: ~
+            uri_factory: ~
+            stream_factory: ~
+
+
+Configuration without auto discovery
+````````````````````````````````````
+
+By default we use Puli to auto discover factories. If you do not want to use auto discovery you could use the following configuration (Guzzle):
```

Comment:
i would say "To avoid the dependency on Puli, you need to specify the client implementation and the 3 factory classes in the ``httplug.classes`` configuration. For Guzzle 6, this would look as follows:"